extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -120,6 +121,16 @@
         public MemorySize maxInput;
     }
 
+    /**
+     * Initialize {@code NTLMEngineImpl} at runtime so as to avoid using a
+     * cached SecureRandom in the image heap.
+     */
+    @BuildStep
+    public RuntimeInitializedClassBuildItem runtimeInitNTLMEngineImpl() {
+        return new RuntimeInitializedClassBuildItem(
+                "org.apache.http.impl.auth.NTLMEngineImpl");
```

Comment:
Hmmm. I thought about it a bit more and this needs to be marked as runtime for extensions using http-client.

This is the case of quarkus-rest-client but not the rest of the RESTEasy stack so I think this should probably be pushed to the quarkus-rest-client-deployment module instead.

But... the problem is that we have several extensions using the Apache HTTP client so I wonder if we will have to create a HTTP client extension just for that... Typically the Elasticsearch REST Client uses it too.

So I think we will have to create an extension for that, unfortunately.

I will try to work on that tomorrow.

Reply:
Makes sense, yes. This needs to be in every possible stack using httpcomponents-client.